testing/api/test_utils_profiling.py

```diff
@@ -63,7 +63,8 @@ def test_timeit(false_atexit, caplog):
 
     # Confirm the reported runtime is ~1 second
     prog_runtime = float(most_recent_log.message.split("; ")[-1].split(' ')[0])
-    assert prog_runtime == pytest.approx(sleep_time, 0.25)
+    allowed_margin = 0.15
+    assert prog_runtime == pytest.approx(sleep_time+allowed_margin, allowed_margin)
```

Comment:
I think I might be getting tripped up over `pytest.approx`'s args including both `rel` and `abs`? 

If 0.15 is supposed to be 0.15s, then I think we'd want:

```python
allowed_margin = 0.15  # Seconds
assert prog_runtime == pytest.approx(sleep_time + allowed_margin, abs=allowed_margin)
```

But if 0.15 is supposed to be 15%, then I think we'd want:

```python
allowed_margin = 0.15  # Percentage
assert prog_runtime == pytest.approx(sleep_time * (1+allowed_margin), rel=allowed_margin)
```

I think the result is the same for a time of 1.00s, but might still be worth clarifying. :stuck_out_tongue_closed_eyes: 

Also, since shifting the `expected` time by `0.15` results in a range of `[1.00, 1.30]`, then maybe we could clarify the intent a little, too:

```python
allowed_margin = 0.30  # Percentage -> [1.00, 1.00 + margin]
assert prog_runtime == pytest.approx(sleep_time * (1+allowed_margin/2), rel=allowed_margin/2)
```

Reply:
Fair point; to make it clearer at-a-glance, I think I'll just split it into two checks:

```suggestion
    allowed_margin = 0.30
    assert prog_runtime >= sleep_time
    assert prog_runtime < sleep_time + allow_margin
```

Thoughts?